### PR TITLE
Simplify boolean expression check

### PR DESCRIPTION
**Pull request by adireddy**
- new `SimplifyBooleanExpression` check

Covers boolean expressions like `if (b == true)`, `b || true`, `!false`, etc.

@Gama11 @AlexHaxe

### CODE REVIEW

File: checkstyle/checks/SimplifyBooleanExpressionCheck.hx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+		for (token in acceptableTokens) {
+			if (token.is(Kwd(KwdTrue)) || token.is(Kwd(KwdFalse))) check(token);
+		}
+	}
+
+	function check(token:TokenTree) {
+		var parent = token.parent;
+		if (parent.is(Binop(OpEq)) || parent.is(Binop(OpNotEq)) || parent.is(Unop(OpNot)) ||
+		parent.is(Binop(OpOr)) || parent.is(Binop(OpAnd)) || parent.is(Binop(OpBoolOr)) ||
```
Comment (Gama11) [inline checkstyle/checks/SimplifyBooleanExpressionCheck.hx]: Missing indent here?

Comment (adireddy) [inline checkstyle/checks/SimplifyBooleanExpressionCheck.hx]: I added indent but intelliJ auto format is removing it... :)

Manually added it now.

**Comment by Gama11** [comment]
This check fails on this code:

``` haxe
@:enum
abstract Test(Int) {}
```

```
./source/Test.hx:1: character 0 : Error: Check SimplifyBooleanExpression failed: bad token Kwd(KwdEnum) != Const(CIdent(_))
```

**Comment by Gama11** [comment]
Haven't spotted any false positives. Pretty useful check, definitely one of the ones I'm gonna put to use. :)

**Comment by AlexHaxe** [comment]
@Gama11 that is a bug in tree tokenizer on dev branch, works on token tree branch

**Comment by Gama11** [comment]
Ah, I see. Are there scenarios in which the token tree branch has bugs that aren't present in current dev? If not the overhead of managing two separate branches might not be worth it, especially if it gets rid of some bugs.

**Comment by adireddy** [comment]
@AlexHaxe is it easy merge tree tokenizer bug fixes into `dev` branch?

**Comment by adireddy** [comment]
I will also wrap checks using tree tokenizer in try catch until we fully merge that branch into dev. Just to avoid any unforeseen errors.

**Comment by AlexHaxe** [comment]
I am not sure if I can cherry pick changes to `checkstyle/Token*`from one branch to another.

Wrapping with a try/catch is a good idea, maybe we should use some kind of fallback system, where we disable token tree checks if one fails, but still run AST and line based checks. And if AST fails we can still run line based checks on that file.

**Comment by adireddy** [comment]
fallback system sounds good

**Comment by AlexHaxe** [comment]
will that cause problems when merging token tree branch?

**Comment by adireddy** [comment]
package names will stay the same so shouldn't be an issue. I will do the move in token-tree branch as well later.

I will ping you before I do any changes in token-tree branch

**Comment by AlexHaxe** [comment]
I have no pending changelists, so go ahead.

**Pull request merged**